acceptance_testing.go

```diff
@@ -1002,8 +1026,39 @@ func (a acceptanceTest) isEqualRecords(is *is.I, want, got []Record) {
 }
 
 func (a acceptanceTest) isEqualRecord(is *is.I, want, got Record) {
-	// TODO figure out if we should compare anything else or at least enforce
-	//  some properties for other fields (e.g. not nil)
-	is.Equal(want.Key, got.Key)         // record key did not match (want != got)
-	is.Equal(want.Payload, got.Payload) // record payload did not match (want != got)
+	a.isEqualData(is, want.Key, got.Key)
+	a.isEqualData(is, want.Payload, got.Payload)
+
+	// check metadata fields, if they are there they should match
+	if len(got.Metadata) == 0 {
+		is.Equal(got.Metadata, nil) // if there is no metadata the map should be nil
```

Comment:
// if there is no metadata the map should be nil

Do we want to allow empty maps?

Reply:
My thinking was that, since it's coming from a connector, we may want to give it a bit more freedom, especially since a nil and an empty map mean the same. But I'm strongly on either side.: ) I do agree that for the code in the SDK it's great to have consistency.

Reply:
Let me add another argument for this assertion - enforcing `nil` means we enforce the good practice of not allocating a map if it's not needed, it could shave off a couple of nanoseconds 🙃

Reply:
You never know when a nanosecond will come good! :D